### PR TITLE
Reorg and Tweaks for on-push GitLab Config File

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
- [x] Ready

## Description
Reorganizes the `.gitlab/ci/on-push.yml` file.
- Two empty lines between configuration sections
- Use `section_start` and `section_end` helper functions in the `create-instances` job
- Move execution of the upload of artifacts from the `after_script` section of the `create-instances` job to the `script` section

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [x] 6.0.0
- [x] 6.1.0
- [x] 6.2.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] ~~Tests~~ N/A
- [x] ~~Documentation~~ N/A

### CODE REVIEW

**Comment by avidan-H** [comment]
Adding the `docs-approved` label since there isn't any documentation requiring review.

File: .gitlab/ci/on-push.yml [hunk 1]
```diff
@@ -46,37 +63,46 @@ create-instances:
...
         ZIP_FOLDER=$(mktemp -d)
         python3 ./Tests/Marketplace/zip_packs.py -b 'marketplace-dist' -z $ZIP_FOLDER -a $ARTIFACTS_FOLDER -s $GCS_MARKET_KEY
       fi
+    - section_end "Zip Marketplace Packs"
+    - section_start "Create Instances"
     - |
-      echo "====== Creating instances ======"
       [ -n "${TIME_TO_LIVE}" ] && TTL=${TIME_TO_LIVE} || TTL=300
       python3 ./Tests/scripts/awsinstancetool/aws_instance_tool.py -envType "$IFRA_ENV_TYPE" -timetolive $TTL -outfile "$ARTIFACTS_FOLDER/env_results.json"
-  after_script:
```
Comment (DinaMeylakh) [inline .gitlab/ci/on-push.yml:76]: will this now inherit the default after_script?

**Review by DinaMeylakh — APPROVED** [review]
Just make sure the after_script behaviour after create-instances is as expected. 
And, thank you for this.

File: .gitlab/ci/on-push.yml (hunk 1 above)
Comment (avidan-H) [reply .gitlab/ci/on-push.yml:76]: There is no default `after_script` anymore.

**Pull request merged**